### ISSUE TITLE
The ip address display on the page FlowView is not correct for the first number

### ISSUE THREAD

**Issue by arno-st** [Cacti/plugin_flowview#186]
On the latest dev version, when I try to see the flow (example video traffic) the first digit of the IP address is wrong:
Her4e is an example whend the first number is displayed as 32, but it should 10, the rest is ok, and on the table plugin_flowview_dnscache the value is correct.
It's also correct on flowview_raw_data_xx

![2024-07-31 11_51_43-Clipboard](https://github.com/user-attachments/assets/52448b9a-0889-4da2-95a5-56ee12afd285)

I'm wondering if the issue is because in Hex the 10 is represented as 0x0a, and 32 is a, a little mixup of the conversion maybe ?

**Comment by TheWitness** [comment]
I don't think so.  Cacti stores as varbinary and uses the INET6_* functions, and nothing else.  Show your filter.

**Comment by arno-st** [comment]
![2024-07-31 14_45_30-Clipboard](https://github.com/user-attachments/assets/1d7378b2-c100-458e-8278-66e8c8ae4276)

![2024-07-31 14_45_53-Clipboard](https://github.com/user-attachments/assets/e7aea020-8b74-4e5a-8b08-968958056813)

As for the protcol it's just TCP.

I remember some issue with some INET6_ tha ATON is still returning a string, not a number.
We had a discussion 2-3 years ago about the filter not working with INET6 but working with INET_, 
https://jira.mariadb.org/browse/MDEV-29862

**Comment by TheWitness** [comment]
They've made a huge improvement in 11.4 that unifies everything into an ITNET column type.  Not shared with MySQL I think.

**Comment by TheWitness** [comment]
Look at my new filter processor for IP patterns.  It should work with CIDR syntax's and non-CIDR syntax's.

**Comment by TheWitness** [comment]
In my install, the varbinary works flawlessly.  How are you confirming the value is correct in the database?

**Comment by arno-st** [comment]
I check the Ip address by looking into the DB:
![2024-08-02 07_25_10-Clipboard](https://github.com/user-attachments/assets/c0bfea97-2e27-4104-9cb0-17adbc7e6508)

It's not fun, now the record are displayed correctly:
![2024-08-02 07_22_11-Clipboard](https://github.com/user-attachments/assets/693c7ef3-343f-484c-a7ef-835b6133f164)


So close the case, thanks

**Comment by TheWitness** [comment]
Yea, weird right.  I communicated the following to another wishing to be anonymous Chinese tester today.

"In my little test setup, I have 3 x 4 core desktops/servers with 16 GB of memory, SMT2 enabled with cheep NVMe 1TB M.2 drives.  They are all connected with MaxScale. 

When searching, combined we are pulling 2GB/second from the three seevers.  If I had 40 or 64 core severs with SMT2 enabled, the NVMe's would be maxed out at 10.5GB/second combined.  If they were PCIe4 drives, it would be 24GB/second.  It's stunningly fast.

Today I fix search releabilty for multiple users."

**Comment by arno-st** [comment]
So again data in wrongly displayed !
![2024-08-02 14_30_15-Clipboard](https://github.com/user-attachments/assets/1033bde7-095f-4262-98f4-1fc67e26272c)

And still on the DB it's 10. something.

But I got this error when I do the search :
02/08/2024 14:21:25 - FLOWVIEW ERROR: Unable to process column:dst_addr information for report:10, type:packets. Please open a ticket on GitHub
02/08/2024 14:21:25 - FLOWVIEW ERROR: Unable to process column:dst_addr information for report:10, type:flows. Please open a ticket on GitHub
02/08/2024 14:21:25 - FLOWVIEW ERROR: Unable to process column:dst_addr information for report:10, type:bytes. Please open a ticket on GitHub 

unable to reproduce, it happen once in a while, and I don't think it' related to the maxscale option; but still try to find something

**Comment by arno-st** [comment]
By the way I got this error when I try to edit a filter:

02/08/2024  14:53:51 - CMDPHP PHP ERROR WARNING Backtrace:   (/plugins/flowview/flowview_filters.php[68]:show_filters(),  /plugins/flowview/flowview_filters.php[376]:CactiErrorHandler())
--
02/08/2024 14:53:51 - ERROR PHP WARNING in  Plugin 'flowview':  Undefined array key 3 in file:  /usr/share/cacti/plugins/flowview/flowview_filters.php  on line: 376
 

Don't know if it's the source of the problem, still investigating

**Comment by arno-st** [comment]
So I find that the problem come when you use the option 'Leverage MaxScale to Distribute Query Shards'
as the result of the query (inside the Blob, on parallel_database_query) is wrong.

But I have no clue how to find the exact query to reproduce it.

**Comment by TheWitness** [comment]
Are you running Max Scale?

**Comment by TheWitness** [comment]
> By the way I got this error when I try to edit a filter:
> ## 02/08/2024 14:53:51 - CMDPHP PHP ERROR WARNING Backtrace: (/plugins/flowview/flowview_filters.php[68]:show_filters(), /plugins/flowview/flowview_filters.php[376]:CactiErrorHandler())
> 
> 02/08/2024 14:53:51 - ERROR PHP WARNING in Plugin 'flowview': Undefined array key 3 in file: /usr/share/cacti/plugins/flowview/flowview_filters.php on line: 376  
> 
> Don't know if it's the source of the problem, still investigating

Okay, with this one, you need to be very specific:

1) What report was already saved for this filter?
2) What was the previous sort field (the numeric value from the table).
3) Open that as a separate ticket once you've isolated 1) and 2) from above.

No more on that here.

**Comment by TheWitness** [comment]
> So again data in wrongly displayed ! ![2024-08-02 14_30_15-Clipboard](https://private-user-images.githubusercontent.com/25684899/354594236-1033bde7-095f-4262-98f4-1fc67e26272c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2OTUwNTQsIm5iZiI6MTcyMjY5NDc1NCwicGF0aCI6Ii8yNTY4NDg5OS8zNTQ1OTQyMzYtMTAzM2JkZTctMDk1Zi00MjYyLTk4ZjQtMWZjNjdlMjYyNzJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDE0MTkxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlNzM3NGRhYWVlMTY5MDFjOGFkYjExOGQwMWJmNjAyYzlmNzIwMmZmNjViOGEwZGI5OGM2N2MyNDE1NGYwN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.saUVfmNzQofSnJ-DPR-i1jOpmrvx0wiWXK_vhr9YahE)
> 
> And still on the DB it's 10. something.
> 
> But I got this error when I do the search : 02/08/2024 14:21:25 - FLOWVIEW ERROR: Unable to process column:dst_addr information for report:10, type:packets. Please open a ticket on GitHub 02/08/2024 14:21:25 - FLOWVIEW ERROR: Unable to process column:dst_addr information for report:10, type:flows. Please open a ticket on GitHub 02/08/2024 14:21:25 - FLOWVIEW ERROR: Unable to process column:dst_addr information for report:10, type:bytes. Please open a ticket on GitHub
> 
> unable to reproduce, it happen once in a while, and I don't think it' related to the maxscale option; but still try to find something

This may be related to the previous ticket.  It may have been something transient from an old filter before I changed the means and mechanisms to determine to sort fields.  So, take that to another ticket.  Make sure it's repeatable first though.  We need exact steps.  Was it a new install for example?  Were you editing an old filter or was this a new one?  Screen shots of the filter.

**Comment by TheWitness** [comment]
On the ip address display, what we really need to do is leverage the PHP error_log function to write the IP addresses to a file and see if it's your flow feed that is faulty or the flow_collector.php that is defective.

Was this the src_addr or the dst_addr that was wrong?

**Comment by TheWitness** [comment]
Also, on the flow issues, was this a v5, v9 or IPFIX feed?

**Comment by TheWitness** [comment]
So, do the following.  This is flow_collector.php.  First set your error log to `<base_path>/log/php_errors.log`.  Then follow the example below:

![image](https://github.com/user-attachments/assets/847a3170-a910-4f89-81e0-d6303b5b6445)

Then wait and see what happens...

**Comment by TheWitness** [comment]
Output should be human readable as follows:

![image](https://github.com/user-attachments/assets/a5ffb826-14dc-46ff-8fba-122ec11e525f)

**Comment by arno-st** [comment]
So here is some answer:
1: it's old filter, when I do the upgrade I don't whip out everything.
2: the error it's both src and dst ip address
3: Yes I do have maxscale
4: it's flow V9
5: On flowview_raw and dns_cache table the IP is correct
6: yes the issue can be recreated, I just did the same query of last week
7: I do have the log, with the good ip adresse.

**Comment by TheWitness** [comment]
Okay, if the log has the good ip address, then I just don't know what the issue is.  We insert into MariaDB as:

`INET6_ATON('$src_addr');`

So, take your IP address and test this in MariaDB:

```sql
SELECT INET6_NTOA(INET6_ATON('your.ip.address.withissue'));
```

See if it's correct or not.

**Comment by arno-st** [comment]
yes it's correct !

**Comment by TheWitness** [comment]
So that means that there has to be some issue from time to time with the address coming from Net-Flow, keep writing the source and destination to the error_log().  Also, purge the entry from the DNS cache and when it comes back, you'll know where to find it.

**Comment by arno-st** [comment]
I find a record on the flowview_raw data who look like this:
![2024-08-08 11_55_26-Clipboard](https://github.com/user-attachments/assets/9adf3394-6ff3-4c1d-b3cc-fc9fc13348d5)

But on the DNS cache it's ok. So I don't think that can explain something, but I have many host with the src_addr correctly recorded, but wrong on the gui. 
I keep thinking that's the problem is not on the collector.

And even more, I don't have this problem if I don't use the 'maxscale sharding' option.

And I follow the setting you propose for maxscale.cnf, and I see a huge improvement on the reaction of the GUI in general.

**Comment by arno-st** [comment]
Hmm flowview was using a readwritesplit with master_only config (the one reserved for cacti)
So I change it to use a readwritesplit like you configured inside you maxscale.cnf
And I disable the usage of 'maxscale sharding' option

And Now I have the problem on the display, but still all DB are correct (flush DNS cash, check manually on the flowview_raw)

**Comment by arno-st** [comment]
We can close this one, the problem is solve.
Don't ask how, but now all my IP are displayed properly.

**Comment by TheWitness** [comment]
Cool.  How's the release working?  @bmfmancini is doing a video.

**Comment by arno-st** [comment]
I had some issue with an 'old' filter, it lost the sort order

`
20/08/2024  08:26:15 - CMDPHP PHP ERROR WARNING Backtrace:   (/plugins/flowview/flowview_filters.php[68]:show_filters(),  /plugins/flowview/flowview_filters.php[376]:CactiErrorHandler())
20/08/2024 08:26:15 - ERROR PHP WARNING in  Plugin 'flowview':  Undefined array key "bytes" in file:  /usr/share/cacti/plugins/flowview/flowview_filters.php  on line: 376
20/08/2024 08:26:15 - CMDPHP PHP ERROR WARNING Backtrace:   (/plugins/flowview/flowview_filters.php[68]:show_filters(),  /plugins/flowview/flowview_filters.php[376]:CactiErrorHandler())
20/08/2024 08:26:15 - ERROR PHP WARNING in  Plugin 'flowview':  Undefined array key "bytes" in file:  /usr/share/cacti/plugins/flowview/flowview_filters.php  on line: 376
20/08/2024 08:26:07 - CMDPHP PHP ERROR WARNING Backtrace:   (/plugins/flowview/flowview_filters.php[68]:show_filters(),  /plugins/flowview/flowview_filters.php[376]:CactiErrorHandler())
20/08/2024 08:26:07 - ERROR PHP WARNING in  Plugin 'flowview':  Undefined array key "bytes" in file:  /usr/share/cacti/plugins/flowview/flowview_filters.php  on line: 376
20/08/2024 08:26:07 - CMDPHP PHP ERROR WARNING Backtrace:   (/plugins/flowview/flowview_filters.php[68]:show_filters(),  /plugins/flowview/flowview_filters.php[376]:CactiErrorHandler())
20/08/2024 08:26:07 - ERROR PHP WARNING in  Plugin 'flowview':  Undefined array key "bytes" in file:  /usr/share/cacti/plugins/flowview/flowview_filters.php  on line: 376
20/08/2024 08:26:05 - CMDPHP PHP ERROR WARNING Backtrace:   (/plugins/flowview/flowview_filters.php[68]:show_filters(),  /plugins/flowview/flowview_filters.php[376]:CactiErrorHandler())
20/08/2024 08:26:05 - ERROR PHP WARNING in  Plugin 'flowview':  Undefined array key "bytes" in file:  /usr/share/cacti/plugins/flowview/flowview_filters.php  on line: 376
20/08/2024 08:26:05 - CMDPHP PHP ERROR WARNING Backtrace:   (/plugins/flowview/flowview_filters.php[68]:show_filters(),  /plugins/flowview/flowview_filters.php[376]:CactiErrorHandler())
20/08/2024 08:26:05 - ERROR PHP WARNING in  Plugin 'flowview':  Undefined array key "bytes" in file:  /usr/share/cacti/plugins/flowview/flowview_filters.php  on line: 376
 `

I just edited my filter and validate the report end the sort field.
So far it's working nicelly, waiting on your feature to display something when working in the background ;)
